pkg/services/object/server.go

```diff
@@ -1147,6 +1168,9 @@ func continueGetFromRemoteNode(ctx context.Context, conn *grpc.ClientConn, req *
 				if !headWas {
 					return io.ErrUnexpectedEOF
 				}
+				if !bytes.Equal(x.payloadHashGot.Sum(nil), x.payloadHashCheck) {
+					return errors.New("received payload mismatches checksum from header")
```

Comment:
IIUC it's too late at this stage. And not sure we can improve anything. Maybe this check can be left for client to perform.

Reply:
considering that client from SDK does checks by default now, I suggest not adding them to server for now